app/views/studies/_expression_file_fields.html.erb

```diff
@@ -19,13 +19,13 @@
   <% current_values = f.object.associated_matrix_files(:raw)&.map { |sf| { label: sf.name, value: sf.id.to_s } } %>
 
   // render the raw counts association select; if replace is true, allows swapping out units dropdown in raw counts form
-  updateRawCountsAssnSelect('#study-file-<%= f.object.id.to_s %>', <%= current_values.to_json.html_safe %>)
+  updateRawCountsAssnSelect('#study-file-<%= f.object.id.to_s %>', <%= current_values.to_json.html_safe %>, <%= User.feature_flag_for_instance(current_user, 'raw_counts_required_backend') %>)
 
   $('#study-file-<%= f.object.id.to_s %>').on('updateRawCountsSelect', function() {
-    updateRawCountsAssnSelect('#study-file-<%= f.object.id.to_s %>', <%= current_values.to_json.html_safe %>)
+    updateRawCountsAssnSelect('#study-file-<%= f.object.id.to_s %>', <%= current_values.to_json.html_safe %>, <%= User.feature_flag_for_instance(current_user, 'raw_counts_required_backend') %>)
   })
 
   $('#study-file-<%= f.object.id.to_s %>').on('change', '.is_raw_counts_true, .is_raw_counts_false', function() {
-    updateRawCountsAssnSelect('#study-file-<%= f.object.id.to_s %>', <%= current_values.to_json.html_safe %>)
+    updateRawCountsAssnSelect('#study-file-<%= f.object.id.to_s %>', <%= current_values.to_json.html_safe %>, <%= User.feature_flag_for_instance(current_user, 'raw_counts_required_backend') %>)
   })
```

Comment:
For easier readability and maintainability, interpolated values for:

* `'#study-file-<%= f.object.id.to_s %>'`
* `<%= current_values.to_json.html_safe %>`
* `<%= User.feature_flag_for_instance(current_user, 'raw_counts_required_backend') %>`

could each be assigned to respective JS variables, and reused.

Reply:
I totally agree - that had been the original implementation, but for some reason it doesn't work.  The select inputs do not update without a page refresh, or re-rendering them by changing the `file_type` select.  I spent a while trying to debug it and decided that it was out-of-scope for this particular PR.  Given that we are actively talking about redoing many of these forms as a part of a larger refactoring, it didn't seem worth holding things up over.